packages/common/src/lib/common/tools.ts

```diff
@@ -2080,23 +2131,27 @@ function sliceArgs(argsObj, startIndex) {
 
 /**
  * Promisifies a function which returns an error as the first argument in its callback
- * @param {Function} fn The function to promisify
- * @param {any} [context=this] (optional) The context (value of `this` to bind the function to)
- * @param {string[]} [returnArgNames] (optional) If the callback contains multiple arguments,
+ * @param fn The function to promisify
+ * @param context (optional) The context (value of `this` to bind the function to)
+ * @param returnArgNames (optional) If the callback contains multiple arguments,
  * you can combine them into one object by passing the names as an array.
  * Otherwise the Promise will resolve with an array
- * @returns {(...args: any[]) => Promise<any>}
  */
-function promisify(fn, context, returnArgNames) {
+function promisify(
```

Comment:
We should kill these functions at some point. I think Node.js is far enough now that we don't need them anymore.

Reply:
We use them in adapter.js a lot currently